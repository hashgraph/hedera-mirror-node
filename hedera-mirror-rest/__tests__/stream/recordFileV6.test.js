/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

import RecordFileV6 from '../../stream/recordFileV6';
import testUtils from './testUtils';

describe('unsupported record file version', () => {
  testUtils.testRecordFileUnsupportedVersion([1, 2, 3, 4, 5], RecordFileV6);
});

describe('canCompact', () => {
  const testSpecs = [
    [1, false],
    [2, false],
    [3, false],
    [4, false],
    [5, false],
    [6, true],
  ];

  testUtils.testRecordFileCanCompact(testSpecs, RecordFileV6);
});

describe('from v6 buffer or compact object', () => {
  testUtils.testRecordFileFromBufferOrObj(6, RecordFileV6, true, true);
});
