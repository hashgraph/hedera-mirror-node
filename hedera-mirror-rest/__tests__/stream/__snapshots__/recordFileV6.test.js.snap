// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`from v6 buffer or compact object check individual field from v6 buffer - toCompactObject - ["0.0.1-123-123456789"] 1`] = `"Transaction 0.0.1-123-123456789 not found in the successful transactions map"`;

exports[`from v6 buffer or compact object check individual field from v6 compact object - toCompactObject - ["0.0.1-123-123456789"] 1`] = `"Transaction 0.0.1-123-123456789 not found in the successful transactions map"`;

exports[`from v6 buffer or compact object end running hash mismatch 1`] = `"End object running hash mismatch"`;

exports[`from v6 buffer or compact object truncated v6 buffer 1`] = `"index out of range: 6658 + 1 > 6658"`;

exports[`from v6 buffer or compact object v6 buffer with extra data 1`] = `"index out of range: 6660 + 1 > 6660"`;

exports[`unsupported record file version create object from version 1 1`] = `"Unsupported record file version, expect 6"`;

exports[`unsupported record file version create object from version 2 1`] = `"Unsupported record file version, expect 6"`;

exports[`unsupported record file version create object from version 3 1`] = `"Unsupported record file version, expect 6"`;

exports[`unsupported record file version create object from version 4 1`] = `"Unsupported record file version, expect 6"`;

exports[`unsupported record file version create object from version 5 1`] = `"Unsupported record file version, expect 6"`;
